src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -865,16 +865,49 @@ public void waitForSchema(int delay)
             }
             Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
         }
-        // if our schema hasn't matched yet, wait until it has
-        // we do this by waiting for all in-flight migration requests and responses to complete
-        // (post CASSANDRA-1391 we don't expect this to be necessary very often, but it doesn't hurt to be careful)
-        if (!MigrationManager.isReadyForBootstrap())
+
+        //if we don't have the same schema as the rest of the live nodes, send new schema pull requests
+        while (!checkForSchemaAgreement())
```

Comment:
do we need to consider any circuit breaker for this while loop?

we could be here forever if schema agreement never happens